docs/examples/use_cases/tensorflow/yolov4/src/dali/ops.py

```diff
@@ -0,0 +1,158 @@
+
+# Based on https://github.com/AlexeyAB/darknet/blob/005513a9db14878579adfbb61083962c99bb0a89/src/image.c#L1297
+def color_twist(images):
+    def random_value():
+        value = dali.fn.random.uniform(range=(1, 1.5))
+        coin = dali.fn.random.coin_flip()
+        return coin * value + (1.0 - coin) * (1.0 / value)
+
+    return dali.fn.color_twist(images,
+        hue=dali.fn.random.uniform(range=(-18.0, 18.0)),
+        brightness=random_value(),
+        contrast=random_value()
+    )
+
+def flip(images, bboxes):
+    coin = dali.fn.random.coin_flip()
+    images = dali.fn.flip(images, horizontal=coin)
+    bboxes = dali.fn.bb_flip(bboxes, horizontal=coin, ltrb=True)
+    return images, bboxes
+
+
+# Performs mosaic using MultiPaste operator.
+def mosaic(images, bboxes, labels, image_size):
+    def generate_tiles(bboxes, labels, shape_x, shape_y):
+        idx = dali.fn.batch_permutation()
+        permuted_boxes = dali.fn.permute_batch(bboxes, indices=idx)
+        permuted_labels = dali.fn.permute_batch(labels, indices=idx)
+        shape = dali.fn.stack(shape_y, shape_x)
+        in_anchor, in_shape, bbx, lbl = dali.fn.random_bbox_crop(
+            permuted_boxes,
+            permuted_labels,
+            input_shape=image_size,
+            crop_shape=shape,
+            shape_layout="HW",
+            allow_no_crop=False,
+            total_num_attempts=64
+        )
+
+        in_anchor = dali.fn.stack(dali.fn.reductions.sum(in_anchor), dali.fn.reductions.sum(in_anchor)) - in_anchor
+        in_anchor_c = dali.fn.cast(in_anchor, dtype=dali.types.DALIDataType.INT32)
+
+        return idx, bbx, lbl, in_anchor_c, shape
+
+    prop0_x = dali.fn.random.uniform(range=(0.2, 0.8))
+    prop0_y = dali.fn.random.uniform(range=(0.2, 0.8))
+    prop1_x = 1.0 - prop0_x
+    prop1_y = 1.0 - prop0_y
+
+    pix0_x = dali.fn.cast(prop0_x * image_size[0], dtype=dali.types.DALIDataType.INT32)
+    pix0_y = dali.fn.cast(prop0_y * image_size[1], dtype=dali.types.DALIDataType.INT32)
+    pix1_x = image_size[0] - pix0_x
+    pix1_y = image_size[1] - pix0_y
+
+    perm_UL, bboxes_UL, labels_UL, in_anchor_UL, size_UL = \
+        generate_tiles(bboxes, labels, pix0_x, pix0_y)
+    perm_UR, bboxes_UR, labels_UR, in_anchor_UR, size_UR = \
+        generate_tiles(bboxes, labels, pix1_x, pix0_y)
+    perm_LL, bboxes_LL, labels_LL, in_anchor_LL, size_LL = \
+        generate_tiles(bboxes, labels, pix0_x, pix1_y)
+    perm_LR, bboxes_LR, labels_LR, in_anchor_LR, size_LR = \
+        generate_tiles(bboxes, labels, pix1_x, pix1_y)
+
+    zeros_i = dali.types.Constant(0)
+    zeros_f = dali.types.Constant(0.0)
+
+    idx = dali.fn.stack(perm_UL, perm_UR, perm_LL, perm_LR)
+    out_anchors = dali.fn.stack(
+        dali.fn.stack(zeros_i, zeros_i),
+        dali.fn.stack(zeros_i, pix0_x),
+        dali.fn.stack(pix0_y, zeros_i),
+        dali.fn.stack(pix0_y, pix0_x)
+    )
```

Comment:
I think for that kind of usage we could use a Python-only function that would get rid of all the inner stacks. That's just thinking aloud and definitely not in the scope of this PR.